### PR TITLE
Adds support for emp3-mirror-cache - feed injection

### PR DESCRIPTION
**Pull request by ee11cbb19052e40b07aac0ca060c23ee**
see #68 

* [core] Disabled previous mirrorcache functionality.
* [api] Added mil.emp3.api.MirrorCache to support integrations with emp3-mirror-cache services.
* [api] Overloaded EMP_Exception constructor to accept a Throwable.
* [test] Updated test-basic to enable testing of MC features.
* [test] Added input dialog when plot testing 'Many Units'.
* [build] Moved travis scripts to gradle directory.
* [build] Removed some 'empty' projects from appearing in IDE.
* [build] Unlinked MirrorCache projects from build; updated dependencies accordingly.
* [build] Removed projects from build: test-validator, test-mirrorcache-standout, test-mirrorcache-joystick, test-mirrorcache-echo.

### CODE REVIEW

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 1]
```diff
@@ -1519,6 +1531,175 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+                    mc.removeProduct(MainActivity.this.oRootOverlay, false);
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_addproduct).setEnabled(true);
+
+                } catch (EMP_Exception e) {
+                    Log.e(TAG, e.getMessage(), e);
+                    new AlertDialog.Builder(MainActivity.this).setTitle("ERROR:").setMessage(e.getMessage()).create().show();
+                    MainActivity.this.oMenu.findItem(R.id.action_mirrorcache_deleteproduct).setEnabled(false);
+                }
+                return true;
+            }
```
Comment (empbarney) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1702]: Why aren't you updating the capability Tester? How is this going to be integration tested?

**Review by Ish-rivera — APPROVED** [review]

**Comment by empbarney** [comment]
162 files? It looks like this was used to remove the validator also?

**Comment by ee11cbb19052e40b07aac0ca060c23ee** [comment]
@empbarney Yes, among other projects. From the notes above:
> [build] Removed projects from build: test-validator, test-mirrorcache-standout, test-mirrorcache-joystick, test-mirrorcache-echo.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java (hunk 1 above)
Comment (ee11cbb19052e40b07aac0ca060c23ee) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1702]: Not knowing how to code for test-vandv, it was much easier. I plan to write integration tests which will get executed during the build. I will also look into updating test-vandv to test MC functionality.

Comment (empbarney) [reply tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1702]: There are instructions in test-vandv. Take a look at AddRemoveGetTest.java class level comments.

**Pull request merged**